### DISCUSSION TITLE
Advice on filtering

### DISCUSSION THREAD

**Discussion by bennythemink** [pocketbase/pocketbase#5167] [Q&A] [answered]
Hi everyone,

I need some advice on filtering. I'm unsure if what I want to do is possible on the server side, and I believe I may have to do the filtering on the client side but I wanted to check first with the community for any advice. 

I have three collections: Courses, Units, and CourseUnits. The Units collection has a field called 'title'. CourseUnits has a field called 'course' and another called 'unit'. These are relation fields to their corresponding collection. A course can have many units and a unit can belong to many courses, with other information stored such as date the connection was made, etc. in the CourseUnits collection.

On a page I present a table of all the units titles in the selected course. The Pocketbase code for this is below. I wish to allow the user to be able to filter the returned list by the unit title.

`export async function getCourseUnitsPaginated(courseId: string, page: number, limit: number, query = '') {
  let options = { sort: '', order: 'desc', expand: 'unit', filter: 'course = "' + courseId + '"' }
  return await pb.collection(Collections.CourseUnits).getList(page, limit, options)
}`

Do I have to make this two steps? First get the full list of units for the selected course, and then filter them on the client side? Or is there some other way to do it, potentially using the "_via_" route? 

Many thanks for any advice.

**Comment by ganigeorgiev** [comment, marked as answer]
If you have a direct relation from the `CourseUnits` collection to `Units` then you can use directly dot-notation to target the unit's title, something like:

```js
filter: "unit.title = 'example'"
// OR if you want to bind as parameter
// (see https://github.com/pocketbase/js-sdk?tab=readme-ov-file#binding-filter-parameters)
filter: pb.filter("unit.title={:title}", { title: "example" })
```

**Reply by ganigeorgiev** [reply]
Another option if you want to list directly the units, could look something like:

```js
await pb.collection(Collections.Units).getList(page, limit, {
    filter: pb.filter("title={:title} && CourseUnits_via_unit.course={:courseId}", {
        title: "example",
        courseId: courseId,
    }),
}) }
```
_Side-note: there is no "order" query parameter. If you want to specify sort ordering you can do that by prepending `+`(default)/`-` to the sort field, ex. `sort:"-title"`._

**Reply by bennythemink** [reply]
Fantastic, thank you Gani. I am away from my computer for 24 hours but will try these once I get back online. (I knew I should have brought my computer to the airport with me lol). 
I will report back once I do and mark this as solved :) 

Many thanks.

**Reply by bennythemink** [reply]
Just back online. @ganigeorgiev your first suggestion works perfectly, thank you very much for your help.
